File: boj/20444.py
import math
n, k = map(int, input().split())


def binary_search_one():
    global n, k
    start=0
    end=n//2
    #print(start, end)
    while start<=end:
        mid=(start+end)//2
        res=(n+1-mid)*(mid+1)
        if res > k:
            end=mid-1
        elif res < k:
            start=mid+1
        else:
            return True
    return False

print("YES" if binary_search_one() else "NO")

File: boj/20444.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.*;
import java.util.Map;

public class Main {

    static long n, m;

    public static boolean binarySearch(){
        long start = 0;
        long end = n / 2;

        while(start<=end){
            long mid = (start+end)/2;
            long res = (n-mid+1)*(mid+1);
            //System.out.println(mid + " " + res);
            if(res == m){
                return true;
            }else if(res > m){
                end=mid-1;
            }else{
                start=mid+1;
            }
        }
        return false;
    }

    private static void solution() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[]t = br.readLine().split(" ");
        n=Long.valueOf(t[0]);
        m=Long.valueOf(t[1]);

        System.out.println(binarySearch() ? "YES" : "NO");


    }


    public static void main(String[] args) throws Exception {
        solution();
    }
}

